tensorzero-core/src/db/datasets.rs

```diff
@@ -90,6 +90,10 @@ pub struct ChatInferenceDatapointInsert {
 
     /// If true, this datapoint was manually created or edited by the user.
     pub is_custom: bool,
+
+    /// Hash of the configuration snapshot that created this datapoint. Optional.
+    #[serde(default)]
+    pub snapshot_hash: Option<crate::config::snapshot::SnapshotHash>,
```

Comment:
I think you should give ClickhouseConnectionInfo the `hash`​ so the value is supplied when we construct the query, instead of at the callsite where the caller need to worry about getting the SnapshotHash object.